presto-raptor/src/main/java/com/facebook/presto/raptor/storage/ShardRecoveryManager.java

```diff
@@ -176,13 +180,22 @@ void restoreFromBackup(UUID shardUuid)
         // move to final location
         storageService.createParents(storageFile);
         try {
-            Files.move(stagingFile.toPath(), storageFile.toPath(), ATOMIC_MOVE);
+            // If storage file does not match the backup file, overwrite the storage file
```

Comment:
This all seems overly complex, especially with the `FileAlreadyExistsException` catch below.